### ISSUE TITLE
Worlds made by the customized world generator crash when disabling lava lakes

### ISSUE THREAD

**Issue by gaspoweredpick** [misode/misode.github.io#399]
After loading a Minecraft world using a datapack from the customized worlds generator with no changes except removing lava lakes, the game crashes. Here is the datapack I used to generate the world:

[customized.zip](https://github.com/misode/misode.github.io/files/11940245/customized.zip)

Here is the full crash report:

The game crashed whilst exception generating new chunk
Error: java.lang.IllegalStateException: Feature order cycle found, involved sources: [Reference{ResourceKey[minecraft:worldgen/biome / minecraft:lush_caves]=cnk@6c0b838a}]

```
---- Minecraft Crash Report ----
// I bet Cylons wouldn't have this problem.

Time: 2023-07-03 13:31:05
Description: Exception generating new chunk

java.lang.IllegalStateException: Feature order cycle found, involved sources: [Reference{ResourceKey[minecraft:worldgen/biome / minecraft:lush_caves]=cnk@6c0b838a}]
	at cnu.a(SourceFile:100)
	at ddy.a(SourceFile:102)
	at com.google.common.base.Suppliers$NonSerializableMemoizingSupplier.get(Suppliers.java:183)
	at ddy.a(SourceFile:288)
	at dec.b(SourceFile:108)
	at dec$d.doWork(SourceFile:309)
	at dec.a(SourceFile:252)
	at ahr.a(SourceFile:684)
	at com.mojang.datafixers.util.Either$Left.map(Either.java:38)
	at ahr.a(SourceFile:673)
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150)
	at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482)
	at aht.b(SourceFile:62)
	at bcq.h(SourceFile:91)
	at bcq.a(SourceFile:146)
	at bcq.run(SourceFile:102)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server thread
Stacktrace:
	at cnu.a(SourceFile:100)
	at ddy.a(SourceFile:102)
	at com.google.common.base.Suppliers$NonSerializableMemoizingSupplier.get(Suppliers.java:183)
	at ddy.a(SourceFile:288)
	at dec.b(SourceFile:108)
	at dec$d.doWork(SourceFile:309)
	at dec.a(SourceFile:252)
	at ahr.a(SourceFile:684)
	at com.mojang.datafixers.util.Either$Left.map(Either.java:38)
	at ahr.a(SourceFile:673)
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150)
	at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482)
	at aht.b(SourceFile:62)
	at bcq.h(SourceFile:91)
	at bcq.a(SourceFile:146)
	at bcq.run(SourceFile:102)

-- Chunk to be generated --
Details:
	Location: -1,-1
	Position hash: -1
	Generator: dhn@238f71a3
Stacktrace:
	at ahr.a(SourceFile:684)
	at com.mojang.datafixers.util.Either$Left.map(Either.java:38)
	at ahr.a(SourceFile:673)
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150)
	at java.base/java.util.concurrent.CompletableFuture$Completion.run(CompletableFuture.java:482)
	at aht.b(SourceFile:62)
	at bcq.h(SourceFile:91)
	at bcq.a(SourceFile:146)
	at bcq.run(SourceFile:102)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)

-- Affected level --
Details:
	All players: 0 total; []
	Chunk stats: 625
	Level dimension: minecraft:overworld
	Level spawn location: World: (8,64,8), Section: (at 8,0,8 in 0,4,0; chunk contains blocks 0,-64,0 to 15,319,15), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,-64,0 to 511,319,511)
	Level time: 0 game time, 0 day time
	Level name: New World
	Level game mode: Game mode: creative (ID 1). Hardcore: false. Cheats: true
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Known server brands: vanilla
	Removed feature flags: 
	Level was modded: false
	Level storage version: 0x04ABD - Anvil
Stacktrace:
	at net.minecraft.server.MinecraftServer.a(SourceFile:376)
	at net.minecraft.server.MinecraftServer.n_(SourceFile:324)
	at fyp.e(SourceFile:69)
	at net.minecraft.server.MinecraftServer.w(SourceFile:646)
	at net.minecraft.server.MinecraftServer.a(SourceFile:265)
	at java.base/java.lang.Thread.run(Thread.java:833)

-- System Details --
Details:
	Minecraft Version: 1.20.1
	Minecraft Version ID: 1.20.1
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 17.0.3, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 251805912 bytes (240 MiB) / 671088640 bytes (640 MiB) up to 2147483648 bytes (2048 MiB)
	CPUs: 12
	Processor Vendor: GenuineIntel
	Processor Name: Intel(R) Core(TM) i5-10500H CPU @ 2.50GHz
	Identifier: Intel64 Family 6 Model 165 Stepping 2
	Microarchitecture: unknown
	Frequency (GHz): 2.50
	Number of physical packages: 1
	Number of physical CPUs: 6
	Number of logical CPUs: 12
	Graphics card #0 name: Intel(R) UHD Graphics
	Graphics card #0 vendor: Intel Corporation (0x8086)
	Graphics card #0 VRAM (MB): 1024.00
	Graphics card #0 deviceId: 0x9bc4
	Graphics card #0 versionInfo: DriverVersion=31.0.101.2111
	Graphics card #1 name: NVIDIA GeForce GTX 1650 with Max-Q Design
	Graphics card #1 vendor: NVIDIA (0x10de)
	Graphics card #1 VRAM (MB): 4095.00
	Graphics card #1 deviceId: 0x1f9d
	Graphics card #1 versionInfo: DriverVersion=30.0.15.1165
	Memory slot #0 capacity (MB): 8192.00
	Memory slot #0 clockSpeed (GHz): 3.20
	Memory slot #0 type: DDR4
	Virtual memory max (MB): 21837.80
	Virtual memory used (MB): 10604.94
	Swap memory total (MB): 13824.00
	Swap memory used (MB): 625.74
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Server Running: true
	Player Count: 0 / 8; []
	Data Packs: vanilla, file/customized.zip
	Enabled Feature Flags: minecraft:vanilla
	World Generation: Experimental
	Type: Integrated Server (map_client.txt)
	Is Modded: Probably not. Client jar signature and brand is untouched; Server jar signature and brand is untouched
	Launched Version: 1.20.1
```

I am new to this place so apologies if my formatting is bad or if I'm missing any information.

**Comment by misode** [comment]
It seems like the fix for #391 did not work...

**Comment by misode** [comment]
Hey @gaspoweredpick I have made an attempted fix. I am sadly unable to test this in-game currently, so if you could verify that the fix works I'd appreciate that!
